packages/frontend/core/src/modules/workspace-indexer-embedding/entities/embedding.ts

```diff
@@ -59,6 +66,15 @@ export class Embedding extends Entity {
     this.getEnabled();
     this.getAttachments({ first: COUNT_PER_PAGE, after: null });
     this.getIgnoredDocs();
+    this.uploadingAttachments$.subscribe(() => this.updateMergedAttachments());
+    this.attachments$.subscribe(() => this.updateMergedAttachments());
+    this.updateMergedAttachments();
+  }
+
+  private updateMergedAttachments() {
+    const uploading = this.uploadingAttachments$.value;
+    const uploaded = this.attachments$.value.edges.map(edge => edge.node);
+    this.mergedAttachments$.next([...uploading, ...uploaded].slice(0, 10));
```

Comment:
Why `slice(0, 10)` here？Pagination should be done at the component level

Reply:
merge locals and remotes, the final list is [locals, persists(remotes)].slice(0,10)